### PR TITLE
Add `TorusMesh`

### PR DESCRIPTION
**Pull request by hoontee**
PrimitiveMesh counterpart to `CSGTorus3D`. ~~Required for #59077.~~

image

### CODE REVIEW

**Comment by hoontee** [comment]
Added tangents. All it's missing now is the resource icon.

**Comment by akien-mga** [comment]
What's the rationale for adding this as a primitive mesh? This is the kind of change that requires a proposal to discuss the use case and whether it makes sense to add things to the API.

**Comment by hoontee** [comment]
CSGTorus3D exists and this was intended to replace it internally when removing redundant CSG primitive generation code.

**Comment by hoontee** [comment]
Now that https://github.com/godotengine/godot/pull/59077 is closed this is solely to have a torus primitive. CSG has one already.

File: scene/resources/primitive_meshes.cpp [hunk 1]
```diff
@@ -1592,6 +1592,136 @@ bool SphereMesh::get_is_hemisphere() const {
...
+
+	ClassDB::bind_method(D_METHOD("set_rings", "rings"), &TorusMesh::set_rings);
+	ClassDB::bind_method(D_METHOD("get_rings"), &TorusMesh::get_rings);
+
+	ClassDB::bind_method(D_METHOD("set_ring_segments", "rings"), &TorusMesh::set_ring_segments);
+	ClassDB::bind_method(D_METHOD("get_ring_segments"), &TorusMesh::get_ring_segments);
+
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "inner_radius", PROPERTY_HINT_RANGE, "0.001,1000.0,0.001,or_greater,exp"), "set_inner_radius", "get_inner_radius");
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "outer_radius", PROPERTY_HINT_RANGE, "0.001,1000.0,0.001,or_greater,exp"), "set_outer_radius", "get_outer_radius");
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "rings", PROPERTY_HINT_RANGE, "3,64,1"), "set_rings", "get_rings");
```
Comment (Calinou) [inline scene/resources/primitive_meshes.cpp:1681]: For large and thin toruses, having more than 64 rings can be useful to improve detail:

```suggestion
	ADD_PROPERTY(PropertyInfo(Variant::INT, "rings", PROPERTY_HINT_RANGE, "3,128,1"), "set_rings", "get_rings");
```

**Comment by Calinou** [comment]
> The resource icon is missing because I don't know how to make one.

Save this file as `editor/icons/TorusMesh.svg` then recompile the editor to see changes:

[TorusMesh.svg.zip](https://github.com/godotengine/godot/files/9258445/TorusMesh.svg.zip)

**Comment by hoontee** [comment]
@Calinou Added your suggestions!

**Review by Calinou — APPROVED** [review]

**Review by reduz — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**